File: 백준/Bronze/1934. 최소공배수/최소공배수.py
m=int(input())

for i in range (m):
    a, b = map(int, input().split())
    
    if a==b:
        print(a)
        continue

    if b > a:
        temp = a
        a = b
        b = temp

    #최대공약수 구하기
    c = a - b
    while True:    
        if a%c==0 and b%c==0:
            #print(c)
            break  # c 가 최대공약수면 빠져나옴
        c = c-1

    #최소공배수 구하기
    a1 = int(a/c)
    b1 = int(b/c)
    print(int(a1*b1*c))

File: 백준/Silver/13241. 최소공배수/최소공배수.java
import java.io.*;
import java.util.Arrays;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        long[] input = Arrays.stream(br.readLine().split(" ")).mapToLong(Long::parseLong).toArray();
        long a = input[0];
        long b = input[1];

        if (a == b) {
            System.out.println(a);
            return;
        }
        if (b > a) {    // a가 더 큰 수가 되도록 swap
            long temp = a;
            a = b;
            b = temp;
        }

        long gcd = gcd(a, b);
        long lcm = (a * b) / gcd;

        System.out.println(lcm);
    }

    static long gcd(long a, long b) {
        while (b != 0) {
            long temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}
